.github/actions/check-skip/action.yaml

```diff
@@ -0,0 +1,144 @@
+name: 'Check skips'
+description: 'Check whether some workflows should be skipped'
+inputs:
+  ignore-changes:
+    required: false
+    description: |
+      Directories and files to ignore when checking changes to determine whether functional
+      tests should be skipped. This is an optional input field and it accepts a regex string.
+      The paths should be relative to the repo's root directory.
+      Examples:
+      - "^(README.md|doc/|tests/unit/)"
+    default: "^$"
+  check-lint-unit-skip:
+    required: false
+    description: "Whether to check the possibility of skipping lint and unit tests"
+    default: false
+  check-func-skip:
+    required: false
+    description: "Whether to check the possibility of skipping functional tests"
+    default: false
+outputs:
+  skip-lint-unit-tests:
+    description: "Whether to skip lint and unit tests"
+    value: ${{ steps.skip-lint-unit-tests.outputs.skip }}
+  skip-func-tests:
+    description: "Whether to skip functional tests"
+    value: ${{ steps.skip-func-tests.outputs.skip }}
```

Comment:
I would argue that we keep `skip-lint-unit-tests` and `skip-func-tests`. If we go with `sucessful-last-run` and `core-files-changed` outputs approach, we are delegating the responsibility of determine whether to run tests to the func and lint-unit jobs. The problem is, at that point, we no longer have access to the information of previous run, so we can no longer provide its url (of course we could add an additional output `last-run-url` to pass on this info, but that would be less cleaner and doesn't make much sense)

Reply:
On the other hand, if we want to make this action as generic as possible so it can be used externally by other teams, maybe having `sucessful-last-run`, `core-files-changed`, and `last-run-url` as output is desirable. We can discuss this further later today.